### PR TITLE
Feature/17+16 student info components

### LINKED ISSUE #3: Admin gränssitt
Lägga till lärare, elever samt tilldela elever till lärare.

### LINKED ISSUE #16: StudentInfoComponent
**DoD**
- Display the email of the user.
- Display the username of the user.
- Render a dropdown for choosing a mentor.

Screenshot 2019-11-18 at 15 34 29

### LINKED ISSUE #17: StudentInfoListComponent
**DoD**
- Display Title "Students"
- Align subtitles(email, username, teacher) with all StudentInfoComponents.
- Render StudentInfoComponents for all students.

Screenshot 2019-11-18 at 15 38 32
"

### LINKED ISSUE #18: AddStudentComponent
**DoD**
- Title "Add new student"
- Render a form that allows the user to send student info (email, username, mentor, password) to the student creation function in the User Module.

Screenshot 2019-11-18 at 15 43 41

### PR DESCRIPTION
**Pull request by te4-gabriel-stark**
## What issue did you implement or fix?
Fixed #17 
Fixed #16 
Fixed #3 
Fixed #18 

## Summary
- Added a StudentInfoComponent that shows you the info of all the students and you can change their main teacher and their password.
- Created the AdminView and imported the StudentInfoComponent to it.
- Created CreateStudentFormComponent where you can create/add a new student to the thing.
- I realised that the StudentInfoListComponent was unnecessary so i deleted it and did it in the StudentInfoComponent instead.

## Type of change
- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

## How can this implementation be improved?
- Add routes in the backend so that you can post the changes from StudentInfoComponent to the database, same for creating a new student/user.
- Import the TeacherInfoComponent to the AdminView aswell.
- Some minor design things that i couldn't figure out.

### CODE REVIEW

File: frontend/src/store/modules/UserModule.ts [hunk 1]
```diff
@@ -2,31 +2,83 @@ import { VuexModule, Module, getModule, Action, Mutation } from 'vuex-module-dec
 import store from '@/store';
 import jwt from 'jsonwebtoken';
 import axios from 'axios';
+import Vue from 'vue';
+import { networkInterfaces } from 'os';
```
[the remark this answers is not in the document]
Comment (te4-gabriel-stark) [reply frontend/src/store/modules/UserModule.ts]:
```suggestion
```

vet faktiskt inte, den bara var där :)

File: frontend/src/components/CreateStudentFormComponent.vue [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+export default class CreateStudentFormComponent extends Vue {
+    private userModule = UserModule;
+    private name: string = "";
+    private email: string = "";
+    private password: string = "";
+    private show: boolean = false;
+    private mentorId: number | null = null;
+
+    createStudent() {
+        UserModule.createStudent({name: this.name, email: this.email, premissons: 0, mentorId: this.mentorId, newPwd: this.password});
```
Comment (TE4-emanuel-strom) [inline frontend/src/components/CreateStudentFormComponent.vue]:
```suggestion
        UserModule.createStudent({name: this.name, email: this.email, permisions: 0, mentorId: this.mentorId, newPwd: this.password});
```

File: frontend/src/store/modules/UserModule.ts [hunk 3]
```diff
@@ -2,31 +2,87 @@ import { VuexModule, Module, getModule, Action, Mutation } from 'vuex-module-dec
...
 }
 
-interface IUser {
-    id: string,
-    username: string
+export interface IUser {
+    id?: string,
+    name: string,
+    email: string,
+    premissons: number | null,
```
Comment (TE4-emanuel-strom) [inline frontend/src/store/modules/UserModule.ts]:
```suggestion
    permisions: number | null,
```

File: frontend/src/store/modules/UserModule.ts [hunk 4]
```diff
@@ -2,31 +2,87 @@ import { VuexModule, Module, getModule, Action, Mutation } from 'vuex-module-dec
...
+    [id: string] : IUser
 }
 
 @Module({ dynamic: true, namespaced: true, name: "UserModule", store: store, preserveState: true })
 class UserModule extends VuexModule {
 
     public isLoading: boolean = false;
     public failure: string = "";
-    public currentUser: IUser = { id: "", username: "" };
+    public currentUser: IUser = { id: "", name: "", email: "", premissons: null, mentorId: null, newPwd: "" };
```
Comment (TE4-emanuel-strom) [inline frontend/src/store/modules/UserModule.ts]:
```suggestion
    public currentUser: IUser = { id: "", name: "", email: "", permisions: null, mentorId: null, newPwd: "" };
```

File: frontend/src/store/modules/UserModule.ts [hunk 5]
```diff
@@ -76,9 +132,85 @@ class UserModule extends VuexModule {
     logout() {
         if (!this.isLoading) {
             this.setAccessToken("");
-            this.setCurrentUser({ id: "", username: "" });
+            this.setCurrentUser({ id: "", name: "" , email: "", premissons: null, newPwd: "", mentorId: null});
```
Comment (TE4-emanuel-strom) [inline frontend/src/store/modules/UserModule.ts]:
```suggestion
            this.setCurrentUser({ id: "", name: "" , email: "", permisions: null, newPwd: "", mentorId: null});
```

**Pull request merged**